src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1221 @@
+}
+
+void 
+DebugIpStatusCollector::getStreamName(uint8_t dbgIpType,
+                        std::string& dbgIpName, 
+                        std::string& masterName, 
+                        std::string& slaveName)
+{
+  // Slotnames are of the format "Master-Slave", split them and store in masterName and slaveName
+  size_t sepPos = dbgIpName.find(IP_LAYOUT_SEP, 0);
+  if(sepPos != std::string::npos) {
+    masterName = dbgIpName.substr(0, sepPos);
+    slaveName  = dbgIpName.substr(sepPos+1);
+  } else {
+    masterName = "Unknown";
+    slaveName  = "Unknown";
+  }
+  cuNameMaxStrLen[dbgIpType] = std::max(masterName.length(), cuNameMaxStrLen[dbgIpType]);
+  portNameMaxStrLen[dbgIpType] = std::max(slaveName.length(), portNameMaxStrLen[dbgIpType]);
+}
+
+
+void 
+DebugIpStatusCollector::readAIMCounter(debug_ip_data* dbgIpInfo)
+{
+  // index in results is debugIpNum
+  const uint64_t index = debugIpNum[AXI_MM_MONITOR];
+
+  // Get Debug Ip Name
+  std::string dbgIpName;
+  // Fill up string with 128 characters (padded with null characters)
+  dbgIpName.assign(dbgIpInfo->m_name, sizeof(dbgIpInfo->m_name));
+  // Strip away any extraneous null characters
+  dbgIpName.assign(dbgIpName.c_str());
+
+  // Get Cu and Port Name
+  std::string cuName;
+  std::string portName;
+  getCuNamePortName(dbgIpInfo->m_type, dbgIpName, cuName, portName);
+  cuNames[AXI_MM_MONITOR].emplace_back(cuName);
+  portNames[AXI_MM_MONITOR].emplace_back(portName);
+
+  // read counter values
+  size_t size = 0;
+
+  static const uint64_t aim_offsets[] = {
+    XAIM_SAMPLE_WRITE_BYTES_OFFSET,
+    XAIM_SAMPLE_WRITE_TRANX_OFFSET,
+    XAIM_SAMPLE_READ_BYTES_OFFSET,
+    XAIM_SAMPLE_READ_TRANX_OFFSET,
+    XAIM_SAMPLE_OUTSTANDING_COUNTS_OFFSET,
+    XAIM_SAMPLE_LAST_WRITE_ADDRESS_OFFSET,
+    XAIM_SAMPLE_LAST_WRITE_DATA_OFFSET,
+    XAIM_SAMPLE_LAST_READ_ADDRESS_OFFSET,
+    XAIM_SAMPLE_LAST_READ_DATA_OFFSET
+  };
+
+  static const uint64_t aim_upper_offsets[] = {
+    XAIM_SAMPLE_WRITE_BYTES_UPPER_OFFSET,
+    XAIM_SAMPLE_WRITE_TRANX_UPPER_OFFSET,
+    XAIM_SAMPLE_READ_BYTES_UPPER_OFFSET,
+    XAIM_SAMPLE_READ_TRANX_UPPER_OFFSET,
+    XAIM_SAMPLE_OUTSTANDING_COUNTS_UPPER_OFFSET,
+    XAIM_SAMPLE_LAST_WRITE_ADDRESS_UPPER_OFFSET,
+    XAIM_SAMPLE_LAST_WRITE_DATA_UPPER_OFFSET,
+    XAIM_SAMPLE_LAST_READ_ADDRESS_UPPER_OFFSET,
+    XAIM_SAMPLE_LAST_READ_DATA_UPPER_OFFSET
+  };
+
+
+  uint32_t currData[XAIM_DEBUG_SAMPLE_COUNTERS_PER_SLOT];
+
+  uint32_t sampleInterval;
+  // Read sample interval register to latch the sampled metric counters
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+  size += xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON,
+                    dbgIpInfo->m_base_address + XAIM_SAMPLE_OFFSET,
+                    &sampleInterval, sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
```

Comment:
This pattern makes the code look ugly.  Please wrap xclRead in an inline function in unnamed namespace at beginning of this file, then call from all places below.  Only the inline function need to have the pragmas and code will be cleaner and easier to read.

Reply:
I have added the wrapper. The code now looks so much better. Thanks for the suggestion.